### PR TITLE
[DOC-9767] Docs for changing between Basic and Standard

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-9767] Docs for changing between Basic and Standard

Preview: 
https://deploy-preview-18934--cockroachdb-docs.netlify.app/docs/cockroachcloud/provision-a-cluster-with-terraform.html#change-a-clusters-plan
https://deploy-preview-18934--cockroachdb-docs.netlify.app/docs/cockroachcloud/cloud-api.html#change-a-clusters-plan

### CODE REVIEW

File: src/current/cockroachcloud/provision-a-cluster-with-terraform.md [hunk 1]
```diff
@@ -179,6 +188,54 @@ The `terraform show` command shows detailed information of your cluster resource
 terraform show
 ~~~
 
+## Change a cluster's plan
+
+To change a CockroachDB {{ site.data.products.basic }} cluster's plan to CockroachDB {{ site.data.products.standard }} in place, or to change a CockroachDB {{ site.data.products.standard }} cluster to CockroachDB {{ site.data.products.basic }}, you can use Terraform.
+
+{{site.data.alerts.callout_info}}
+To migrate between CockroachDB {{ site.data.products.advanced }} and either CockroachDB {{ site.data.products.standard }} or CockroachDB {{ site.data.products.basic }}, you must create and configure a new cluster, back up the existing cluster's data, and restore the backup to the new cluster. Migration in place is not supported. Refer to [Customer-owned backups]({% link cockroachcloud/backup-and-restore-overview.md %}#customer-owned-backups).
```
Comment (marksoper) [inline src/current/cockroachcloud/provision-a-cluster-with-terraform.md:196]: @mdlinville let's leave out Basic here - Advanced to Basic isn't a scenario we're planning to support, so customers will need to go Advanced to Standard, and then to Basic in the unlikely event that they would want to transition between Advanced and Basic.

File: src/current/cockroachcloud/provision-a-cluster-with-terraform.md [hunk 2]
```diff
@@ -179,6 +188,54 @@ The `terraform show` command shows detailed information of your cluster resource
...
+1. Apply the template:
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    terraform apply
+    ~~~
+
+To migrate from CockroachDB {{ site.data.products.standard }} to CockroachDB {{ site.data.products.basic }} in place:
+
+1. Edit the cluster's Terraform template:
+    -  Change `plan` to `BASIC`.
```
Comment (marksoper) [inline src/current/cockroachcloud/provision-a-cluster-with-terraform.md:221]: @mdlinville would it work to mention here that downgrading to Basic successfully requires that all Standard-only features be deactivated - e.g. any private endpoints must be deleted.

File: src/current/cockroachcloud/provision-a-cluster-with-terraform.md [hunk 3]
```diff
@@ -179,6 +188,78 @@ The `terraform show` command shows detailed information of your cluster resource
...
+<section class="filter-content" markdown="1" data-scope="terraform">
+
+To migrate from CockroachDB {{ site.data.products.basic }} to CockroachDB {{ site.data.products.standard }} in place:
+
+1. Edit the cluster's Terraform template:
+    - Change `plan` to `STANDARD`.
+    - Replace the contents of `serverless {}` (which may be empty) with the provisioned vCPUs for the cluster. This field is required for CockroachDB {{ site.data.products.standard }}. It is not possible to set storage limitations on CockroachDB {{ site.data.products.standard }}.
+      {% include_cached copy-clipboard.html %}
+      ~~~ hcl
+            serverless = {
```
Comment (mikeCRL) [inline src/current/cockroachcloud/provision-a-cluster-with-terraform.md:213]:
```suggestion
        serverless = {
```
Removing four spaces before serverless for proper alignment

File: src/current/cockroachcloud/provision-a-cluster-with-terraform.md [hunk 4]
```diff
@@ -179,6 +188,78 @@ The `terraform show` command shows detailed information of your cluster resource
...
+    terraform apply
+    ~~~
+
+</section>
+
+<section class="filter-content" markdown="1" data-scope="api">
+
+To migrate from CockroachDB {{ site.data.products.basic }} to CockroachDB {{ site.data.products.standard }} in place:
+
+1. Edit the API command or application code that was used to created the cluster. In the JSON header, replace the contents of `serverless {}` (which may be empty) with the provisioned vCPUs for the cluster. This field is required for CockroachDB {{ site.data.products.standard }}. It is not possible to set storage limitations on CockroachDB {{ site.data.products.standard }}. Refer to [CockroachDB API: Create a Standard cluster](https://cockroachlabs.com/docs/api/cloud/v1.html#post-/api/v1/clusters).
```
Comment (mikeCRL) [inline src/current/cockroachcloud/provision-a-cluster-with-terraform.md:252]:
```suggestion
1. Edit the API command or application code that was used to create the cluster. In the JSON header, replace the contents of `serverless {}` (which may be empty) with the provisioned vCPUs for the cluster. This field is required for CockroachDB {{ site.data.products.standard }}. It is not possible to set storage limitations on CockroachDB {{ site.data.products.standard }}. Refer to [CockroachDB API: Create a Standard cluster](https://cockroachlabs.com/docs/api/cloud/v1.html#post-/api/v1/clusters).
```

Also I like the "H1: H2" link format. Haven't thought to use that in our docs before.

**Review by mikeCRL — APPROVED** [review]

File: src/current/cockroachcloud/provision-a-cluster-with-terraform.md (hunk 3 above)
Comment (mdlinville) [reply src/current/cockroachcloud/provision-a-cluster-with-terraform.md:213]: Done

File: src/current/cockroachcloud/provision-a-cluster-with-terraform.md (hunk 4 above)
Comment (mdlinville) [reply src/current/cockroachcloud/provision-a-cluster-with-terraform.md:252]: We have done this a few times when we are linking to a specific section of a long page, just to make the link text a little more accessible. I applied your suggestion elsewhere.

File: src/current/cockroachcloud/cloud-api.md [hunk 5]
```diff
@@ -387,6 +387,25 @@ Where:
 
 If the request was successful, the client will not receive a response payload.
 
+## Change a cluster's plan
+
+This section shows how to change a cluster's plan using the CockroachDB {{ site.data.products.cloud }} API. To use Terraform instead, refer to [Provision a cluster with Terraform]({% link cockroachcloud/provision-a-cluster-with-terraform#change-a-clusters-plan}).
```
Comment (mikeCRL) [inline src/current/cockroachcloud/cloud-api.md:392]:
```suggestion
This section shows how to change a cluster's plan using the CockroachDB {{ site.data.products.cloud }} API. To use Terraform instead, refer to [Provision a cluster with Terraform]({% link cockroachcloud/provision-a-cluster-with-terraform#change-a-clusters-plan %}).
```

**Review by mikeCRL — APPROVED** [review]
LGTM per the diff. Committed a fix to add a missing ` %}` which will hopefully fix the preview build.

**Review by mikeCRL — APPROVED** [review]

**Pull request merged**